src/python/grpcio/grpc/_cython/_cygrpc/records.pyx.pxi

```diff
@@ -303,21 +304,46 @@ cdef class SslPemKeyCertPair:
     self.c_pair.certificate_chain = self.certificate_chain
 
 
+
+cdef void* copy_ptr(void* ptr):
+  return ptr
+
+
+cdef void destroy_ptr(void* ptr):
+  pass
+
+
+cdef int compare_ptr(void* a, void* b):
+  if a < b:
+    return -1
+  elif a > b:
+    return 1
+  else:
+    return 0
+
+
 cdef class ChannelArg:
 
   def __cinit__(self, bytes key, value):
     self.key = key
+    self.value = value
     self.c_arg.key = self.key
     if isinstance(value, int):
-      self.value = value
       self.c_arg.type = GRPC_ARG_INTEGER
       self.c_arg.value.integer = self.value
     elif isinstance(value, bytes):
-      self.value = value
       self.c_arg.type = GRPC_ARG_STRING
       self.c_arg.value.string = self.value
     else:
-      raise TypeError('Expected int or bytes, got {}'.format(type(value)))
+      # Pointer objects must override __int__() to return
+      # the underlying C address.  This is how SWIG exposes
+      # pointers (Python ints are word size)
```

Comment:
I'll document that the lifetime of the pointer is bound to the lifetime of the object.  Because we save a reference to the python object in the ChannelArg, and Servers/Channels save references to their channel args, I don't think we need to be too worried about object lifetime here.

We can't use Py_INCREF/Py_DECREF in the copy/destroy methods because those methods take the underlying pointer as arguments, not the PyObject*.